components/memix/actions/random/random.mjs

```diff
@@ -0,0 +1,29 @@
+import { axios } from "@pipedream/platform";
+export default {
+  name: "Caption to Memix URL",
+  description:
+    "Generate a Memix share URL with a random template using the supplied caption",
+  key: "memix_random",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    caption: {
+      type: "string",
+      label: "Caption",
+    },
+  },
+  async run() {
+    const data = await axios(this, {
+      method: "get",
+      headers: {
+        "X-API-Partner": "pipedream",
+      },
+      url: "https://partner-api.memix.com/templates",
+    });
```

Comment:
you could move this axios call to the app file creating a function such as `getMemixTemplates()`, and then, call this function here.

Reply:
got it. i did some refactoring to clean it all up